### PR TITLE
publiccloud: Fix install aws-cli from Cloud:Tools

### PR DESCRIPTION
**Pull request by cfconrad**
Use workaround from boo#1122199.

- Related ticket: https://progress.opensuse.org/issues/46250
- Verification run: http://cfconrad-vm.qa.suse.de/tests/3225

### CODE REVIEW

**Comment by cfconrad** [comment]
@jlausuch WDYT

File: tests/publiccloud/prepare_tools.pm [hunk 1]
```diff
@@ -46,9 +46,13 @@ sub run {
         assert_script_run("pip3 install -q keyring");
     }
     elsif (is_sle) {
-        zypper_call('rr Cloud_Tools');
-        zypper_call('ref');
         zypper_call('-q in aws-cli');
+
+        # boo#1122199
+        zypper_call('ar http://download.opensuse.org/repositories/devel:/languages:/python:/aws/' . generate_version() . '/devel:languages:python:aws.repo');
+        zypper_call('--gpg-auto-import-keys -q in -f --repo devel_languages_python_aws python-s3transfer');
```
Comment (asmorodskyi) [inline tests/publiccloud/prepare_tools.pm]: mmm so you adding repo , refreshing it and remove it ? what I missing ?

Comment (asmorodskyi) [reply tests/publiccloud/prepare_tools.pm]: also check `zypper_ar` function for adding repos

Comment (cfconrad) [reply tests/publiccloud/prepare_tools.pm]: I will use `zypper_ar`.
The second one is a `zypper in`.

Comment (asmorodskyi) [reply tests/publiccloud/prepare_tools.pm]: ah yeah now I see , sorry I overlook it

File: tests/publiccloud/prepare_tools.pm [hunk 2]
```diff
@@ -46,9 +46,15 @@ sub run {
         assert_script_run("pip3 install -q keyring");
     }
     elsif (is_sle) {
-        zypper_call('rr Cloud_Tools');
-        zypper_call('ref');
         zypper_call('-q in aws-cli');
+
+        # boo#1122199
```
Comment (okurz) [inline tests/publiccloud/prepare_tools.pm]: I recommend to record a soft-failure for this one

**Comment by jlausuch** [comment]
tested on my end and build works: http://fromm.arch.suse.de/tests/4843
I also tried aws command and they work as before.

File: tests/publiccloud/prepare_tools.pm (hunk 2 above)
Comment (jlausuch) [reply tests/publiccloud/prepare_tools.pm]: this PR actually fixes the problem. Shall we remove the bug report in the comment?

Comment (asmorodskyi) [reply tests/publiccloud/prepare_tools.pm]: and I would recommend to remove the comment. Bug has resolved state and actually was not a bug from my POV

Comment (cfconrad) [reply tests/publiccloud/prepare_tools.pm]: Ok, I will add  it.

File: tests/publiccloud/prepare_tools.pm [hunk 3]
```diff
@@ -46,9 +46,15 @@ sub run {
...
     elsif (is_sle) {
-        zypper_call('rr Cloud_Tools');
-        zypper_call('ref');
         zypper_call('-q in aws-cli');
+
+        record_soft_failure('workaround for boo#1122199');
+        my $repo      = 'http://download.opensuse.org/repositories/devel:/languages:/python:/aws/' . generate_version();
+        my $repo_name = 'devel_languages_python_aws';
+        zypper_ar($repo, $repo_name);
+        zypper_call('--gpg-auto-import-keys -q in -f --repo ' . $repo_name . ' python-s3transfer');
```
Comment (asmorodskyi) [inline tests/publiccloud/prepare_tools.pm]: `zypper_ar` already taking about keys import so you can remove this key

File: tests/publiccloud/prepare_tools.pm (hunk 2 above)
Comment (okurz) [reply tests/publiccloud/prepare_tools.pm]: Now I am also confused. We should use soft-fails *only* for bugs that are open and we want to be fixed. This looks pretty messy now. To my understanding we should use the tools from the offical SLE repos. If we can't then the OBS repos should provide the right dependencies. Maybe you could try to add a multibuild package self-test to aws-cli like I did for azure-cli already in https://build.opensuse.org/request/show/651347 ?

**Comment by cfconrad** [comment]
Run with the latest version: http://cfconrad-vm.qa.suse.de/tests/3237

File: tests/publiccloud/prepare_tools.pm (hunk 2 above)
Comment (okurz) [reply tests/publiccloud/prepare_tools.pm]: ok

**Pull request merged**